controller/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{self, Mint, MintTo, Burn, TokenAccount, Token};
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+#[program]
+pub mod controller {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let controller = &mut ctx.accounts.controller_state;
+
+        controller.token_mint = ctx.accounts.token_mint.key();
+        controller.owner = ctx.accounts.signer.key();
+
+        Ok(())
+    }
+
+    pub fn set_members(ctx: Context<SetMembers>) -> Result<()> {
+        let controller = &mut ctx.accounts.controller_state;
+        controller.members = ctx.accounts.members.key();
+
+        Ok(())
+    }
+
+    pub fn set_factory(ctx: Context<SetFactory>) -> Result<()> {
```

Comment:
why factory is not part of the controller? I'm trying to understand why factory needs to CPI into the controller to mint instead of just implementing the mint instruction?